@navikt/core/react/src/stepper/Stepper.tsx

```diff
@@ -94,41 +94,37 @@ export const Stepper: StepperComponent = forwardRef<
           className,
         )}
       >
-        <StepperContext.Provider
-          value={{
-            activeStep,
-            onStepChange,
-            lastIndex: React.Children.count(children),
-            orientation,
-            interactive,
-          }}
-        >
-          {React.Children.map(children, (step, index) => {
-            return (
-              <li
-                className={cl("navds-stepper__item", {
-                  "navds-stepper__item--behind": activeStep > index,
-                  "navds-stepper__item--completed":
-                    React.isValidElement<StepperStepProps>(step) &&
-                    step?.props?.completed,
-                  "navds-stepper__item--non-interactive":
-                    React.isValidElement<StepperStepProps>(step) &&
-                    !(step?.props?.interactive ?? interactive),
-                })}
-                key={index + (children?.toString?.() ?? "")}
+        {React.Children.map(children, (step, index) => {
+          return (
+            <li
+              className={cl("navds-stepper__item", {
+                "navds-stepper__item--behind": activeStep > index,
+                "navds-stepper__item--completed":
+                  React.isValidElement<StepperStepProps>(step) &&
+                  step?.props?.completed,
+                "navds-stepper__item--non-interactive":
+                  React.isValidElement<StepperStepProps>(step) &&
+                  !(step?.props?.interactive ?? interactive),
+              })}
```

Comment:
I suggest refactoring isValidElement etc.:
```suggestion
          const stepProps: Partial<StepperStepProps> =
            React.isValidElement<StepperStepProps>(step) ? step.props : {};
          return (
            <li
              className={cl("navds-stepper__item", {
                "navds-stepper__item--behind": activeStep > index,
                "navds-stepper__item--completed": stepProps.completed,
                "navds-stepper__item--non-interactive": !(
                  stepProps.interactive ?? interactive
                ),
              })}
```